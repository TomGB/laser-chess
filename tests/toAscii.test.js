const chess = require('../src/')

describe('Laser Chess to ASCII', () => {
  const game = chess.newGame()
  const asciiOutput = chess.toAscii(game)

  it('visualises the board using ascii characters', () => {
    expect(asciiOutput).toMatchSnapshot()
  })
})
