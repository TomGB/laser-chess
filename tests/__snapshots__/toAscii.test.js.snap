// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laser Chess to ASCII visualises the board using ascii characters 1`] = `
"game setup complete
    a b c d e f g h i j

0   [31m⇣ [0m[2m· [0m[2m  [0m[2m  [0m[31m⤩ [0m[31m♔ [0m[31m⤩ [0m[31m◸ [0m[31m[2m· [0m[2m· [0m
1   [31m[2m· [0m[2m  [0m[31m◹ [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m· [0m
2   [31m[2m· [0m[2m  [0m[2m  [0m◿ [2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m· [0m
3   [31m◺ [0m[2m  [0m◹ [2m  [0m[31m⟍ [0m[31m⟋ [0m[2m  [0m[31m◸ [0m[2m  [0m◿ 
4   [31m◸ [0m[2m  [0m◿ [2m  [0m⟋ ⟍ [2m  [0m[31m◺ [0m[2m  [0m◹ 
5   [31m[2m· [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[31m◸ [0m[2m  [0m[2m  [0m[2m· [0m
6   [31m[2m· [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m[2m  [0m◺ [2m  [0m[2m· [0m
7   [31m[2m· [0m[2m· [0m◿ ⤧ ♔ ⤧ [2m  [0m[2m  [0m[31m[2m· [0m⇡ "
`;
